File: umbrella-insurance-react-ui/src/models/teamTransactions/v1/TeamTransaction.ts
import { Team } from "../../teams/v1/Team";
import { TeamTransactionType } from "../teamTransactionTypes/v1/TeamTransactionType";

export class TeamTransaction {
    id?: number;
    dateOfTeamTransaction?: string;
    teamTransactionType?: TeamTransactionType;
    description?: string;
    team?: Team;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.dateOfTeamTransaction) {
                this.dateOfTeamTransaction = obj.dateOfTeamTransaction;
            }
            if(obj?.teamTransactionType) {
                this.teamTransactionType = new TeamTransactionType(obj.teamTransactionType);
            }
            if(obj?.description) {
                this.description = obj.description;
            }
            if(obj?.team) {
                this.team = new Team(obj.team);
            }
        }
    }
}

File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/TeamTransaction.java
package com.umbrella.insurance.core.models.entities;

import com.umbrella.insurance.core.deserializers.DateDeserializer;
import com.umbrella.insurance.core.serializers.DateSerializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.sql.Date;

@Getter
@Setter
@Entity
@Table(name = "team_transactions", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "team_transactions_unique", columnNames = {"team_id", "description"})
})
public class TeamTransaction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "team_transaction_id", nullable = false)
    private Long id;

    @JsonSerialize(using = DateSerializer.class)
    @JsonDeserialize(using = DateDeserializer.class)
    @Column(name = "date_of_team_transaction")
    private Date dateOfTeamTransaction;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "team_transaction_type_id")
    private TeamTransactionType teamTransactionType;

    @NotNull
    @Column(name = "description", nullable = false, length = Integer.MAX_VALUE)
    private String description;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "team_id")
    private Team team;

}